browsergym/experiments/src/browsergym/experiments/benchmark.py

```diff
@@ -0,0 +1,295 @@
+    multiaction: bool
+    strict: bool
+    retry_with_force: bool
+    demo_mode: Literal["off", "default", "all_blue", "only_visible_elements"]
+
+    def make_action_set(self):
+        return HighLevelActionSet(
+            subsets=self.subsets,
+            custom_actions=None,
+            multiaction=self.multiaction,
+            strict=self.strict,
+            retry_with_force=self.retry_with_force,
+            demo_mode=self.demo_mode,
+        )
+
+
+@dataclass
+class Benchmark(DataClassJsonMixin):
+    name: str
+    high_level_action_set: HighLevelActionSetArgs
+    env_args_list: list[EnvArgs]
+
+
+def task_list_from_metadata(benchmark: str, *args, **kwargs):
+    return task_list_from_csv(
+        io.StringIO(
+            pkgutil.get_data(__name__, f"benchmark_metadata/{benchmark}.csv").decode("utf-8")
+        ),
+        *args,
+        **kwargs,
+    )
+
+
+def task_list_from_csv(
+    filepath_or_bytes: str | pathlib.Path | io.IOBase,
+    filters: dict[str, str] = {},
+):
+    # read task list with metadata
+    df: pd.DataFrame = pd.read_csv(filepath_or_bytes)
+    # filter the desired columns (AND filter)
+    for col_name, regex in filters.items():
+        filter = df[col_name].astype(str).str.contains(regex, regex=True)
+        df = df[filter]
+    # return only the task names
+    return df["task_name"]
+
+
+# These are mean as the default highlevel action set to fairly evaluate agents on each benchmark.
+# They are mostly arbitrary, the important thing is to evaluate different agents using the same action set for fairness.
+DEFAULT_HIGHLEVEL_ACTION_SETS = {
+    "miniwob": HighLevelActionSetArgs(
+        subsets=["bid", "coord"],
+        multiaction=False,
+        strict=False,
+        retry_with_force=False,
+        demo_mode="off",
+    ),
+    "miniwob_bid": HighLevelActionSetArgs(
+        subsets=["bid"],
+        multiaction=False,
+        strict=False,
+        retry_with_force=False,
+        demo_mode="off",
+    ),
+    "workarena_l1": HighLevelActionSetArgs(
+        subsets=["chat", "bid"],  # no need for infeasible action
+        multiaction=False,
+        strict=False,
+        retry_with_force=False,
+        demo_mode="off",
+    ),
+    "workarena": HighLevelActionSetArgs(
+        subsets=["chat", "infeas", "bid"],
+        multiaction=False,
+        strict=False,
+        retry_with_force=False,
+        demo_mode="off",
+    ),
+    "webarena": HighLevelActionSetArgs(
+        subsets=["chat", "infeas", "bid", "tab", "nav"],
+        multiaction=False,
+        strict=False,
+        retry_with_force=False,
+        demo_mode="off",
+    ),
+    "visualwebarena": HighLevelActionSetArgs(
+        subsets=["chat", "infeas", "bid", "tab", "nav"],
+        multiaction=False,
+        strict=False,
+        retry_with_force=False,
+        demo_mode="off",
+    ),
+}
+
+BENCHMARKS = {
+    "miniwob_all": lambda: Benchmark(
```

Comment:
put a commont to reassure the surprised user that there is a lambda

Reply:
done